modules/account-lib/test/unit/coin/dot/transactionBuilder/addProxyBuilder.ts

```diff
@@ -0,0 +1,148 @@
+import { coins } from '@bitgo/statics';
+import should from 'should';
+import sinon, { assert } from 'sinon';
+import { AddProxyBuilder } from '../../../../../src/coin/dot';
+import * as DotResources from '../../../../resources/dot';
+
+describe('Dot Add Proxy Builder', () => {
+  let builder: AddProxyBuilder;
+
+  const sender = DotResources.accounts.account1;
+  const receiver = DotResources.accounts.account3;
+
+  beforeEach(() => {
+    const config = coins.get('dot');
+    builder = new AddProxyBuilder(config);
+  });
+  describe('setter validation', () => {
+    it('should validate delay', () => {
+      const spy = sinon.spy(builder, 'validateValue');
+      should.throws(
+        () => builder.delay('-1'),
+        (e: Error) => e.message === 'Value cannot be less than zero',
+      );
+      should.doesNotThrow(() => builder.delay('0'));
+      assert.calledTwice(spy);
+    });
+
+    it('should validate delegate address', () => {
+      const spy = sinon.spy(builder, 'validateAddress');
+      should.throws(
+        () => builder.delegate('asd'),
+        (e: Error) => e.message === `The address 'asd' is not a well-formed dot address`,
+      );
+      should.doesNotThrow(() => builder.delegate(sender.address));
+      assert.calledTwice(spy);
+    });
+  });
+
+  describe('build addProxy transaction', () => {
+    it('should build a addProxy transaction', async () => {
+      builder
+        .testnet()
+        .delegate(receiver.address)
```

Comment:
`delegate` is too coin specific and it is not clear (it can be misunderstood with staking). Rename to `owner`